staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -2198,3 +2212,18 @@ func hasAdditionalNewField(original, modified map[string]interface{}) bool {
 	}
 	return false
 }
+
+func setMergeKeyForEmptyMapInSlice(m []interface{}, mergeKey string) error {
+	for _, v := range m {
+		typedV, ok := v.(map[string]interface{})
+		if !ok {
+			return mergepatch.ErrBadArgType(typedV, v)
+		}
+
+		if len(typedV) == 0 {
+			typedV[mergeKey] = nil
```

Comment:
can typedV be nil?

Reply:
seems can't be nil according to [json.Unmarshal](https://pkg.go.dev/encoding/json#:~:text=If%20the%20map%20is%20nil%2C%20Unmarshal%20allocates%20a%20new%20map), added check for safety